File: src/main/java/br/com/g1bet/model/Partida.java
package br.com.g1bet.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;

@Entity
public class Partida {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_time_visitante", referencedColumnName = "id")
    private Time timeVisitante;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "id_time_casa", referencedColumnName = "id")
    private Time timeCasa;

    @NotNull
    @Size(min = 5, max = 20)
    private String resultado;

    @Column(name = "data_hora_partida")
    private LocalDateTime dataHora;

    public Partida() {

    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Time getTimeVisitante() {
        return timeVisitante;
    }

    public void setTimeVisitante(Time timeVisitante) {
        this.timeVisitante = timeVisitante;
    }

    public Time getTimeCasa() {
        return timeCasa;
    }

    public void setTimeCasa(Time timeCasa) {
        this.timeCasa = timeCasa;
    }

    public LocalDateTime getDataHora() {
        return dataHora;
    }

    public void setDataHora(LocalDateTime dataHora) {
        this.dataHora = dataHora;
    }

    public String getResultado() {
        return resultado;
    }

    public void setResultado(String resultado) {
        this.resultado = resultado;
    }

}

File: src/main/java/br/com/g1bet/model/Partida.kt
package br.com.g1bet.model

import com.fasterxml.jackson.annotation.JsonFormat
import org.jetbrains.annotations.NotNull
import java.time.LocalDateTime
import javax.persistence.*
import javax.validation.constraints.Size

@Entity
class Partida {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @ManyToOne
    @JoinColumn(name = "id_time_visitante", referencedColumnName = "id")
    var timeVisitante: Time? = null

    @ManyToOne
    @JoinColumn(name = "id_time_casa", referencedColumnName = "id")
    var timeCasa: Time? = null

    @NotNull
    @Size(min = 5, max = 20)
    var resultado: String? = null

    @Column(name = "data_hora_partida")
    @JsonFormat(pattern = "dd/MM/yyyy")
    var dataHora: LocalDateTime? = null
}